src/nvim/regexp_nfa.c

```diff
@@ -6068,8 +6082,12 @@ static long nfa_regtry(nfa_regprog_T *prog, colnr_T col)
   clear_sub(&subs.synt);
   clear_sub(&m.synt);
 
-  if (nfa_regmatch(prog, start, &subs, &m) == FALSE)
+  result = nfa_regmatch(prog, start, &subs, &m);
+  if (result == FALSE) {
     return 0;
+  } else if (result == NFA_TOO_EXPENSIVE) {
+    return result;
+  }
```

Comment:
@splinterofchaos Thanks very much for looking this over!

I'm not comfortable touching too many `TRUE`/`FALSE`s at the moment, I have a feeling there are semantic subtleties that I might miss (here it's explicitly returning `0` instead of the `FALSE` result, which would of course be the same since it's `#def`ed to be `0`).

But I will change it if you insist.